src/near_to_eth_objects.js

```diff
@@ -49,41 +49,56 @@ nearToEth.transactionObj = async function(tx, txIndex) {
 
     let destination = null;
     let data = null;
+    let value = null;
     const { transaction_outcome, transaction } = tx;
 
     const functionCall = transaction.actions[0].FunctionCall;
-    // if it's a call, get the destination address
-    if (functionCall && functionCall.method_name == 'call_contract') {
-      const args = JSON.parse(utils.base64ToString(functionCall.args));
-      destination = args.contract_address;
-      data = args.encoded_input;
+    if (functionCall) {
+        const args = JSON.parse(utils.base64ToString(functionCall.args));
+        switch (functionCall.method_name) {
+            case 'call_contract':
+                destination = args.contract_address;
+                data = args.encoded_input;
+                break;
+            case 'deploy_code':
+                data = args.bytecode;
+                break;
+            case 'add_near':
+                destination = utils.nearAccountToEvmAddress(transaction.signer_id);
+                break;
+            case 'move_funds_to_evm_address':
+                destination = args.address;
+                value = parseInt(args.amount)
+                break;
+        }
     }
 
     const sender = utils.nearAccountToEvmAddress(transaction.signer_id);
-    // const receiver = utils.nearAccountToEvmAddress(transaction.receiver_id);
 
-    const value = transaction.actions.map(v => {
-        const k = Object.keys(v)[0];
-        return parseInt(v[k].deposit, 10);
-    }).reduce((a, b) => a + b);
+    if (value == null) {
+        value = transaction.actions.map(v => {
+            const k = Object.keys(v)[0];
+            return parseInt(v[k].deposit, 10);
+        }).reduce((a, b) => a + b);
+    }
 
     const obj = {
         // DATA 20 bytes - address of the sender
         // from: sender,
         from: sender,
 
         // DATA 20 bytes - address of the receiver
-        to: utils.include0x(destination),
+        to: destination ? utils.include0x(destination) : null,
 
         // QUANTITY - integer of the current gas price in wei
         // TODO: This will break with big numbers?
         gasPrice: utils.decToHex(parseInt(tx.gas_price)),
 
         // DATA - the data sent along with the transaction
-        input: '0x' + data ? data : '',
+        input: data ? utils.include0x(data) : '',
 
         // DATA 32 bytes - hash of the block where this transaction was in
-        blockHash: transaction_outcome.block_hash,
+        blockHash: utils.base58ToHex(transaction_outcome.block_hash),
```

Comment:
on second thought, maybe we want to retain the base58 blockhash for nearProtocol

Reply:
on third though, I see we require the hex representation of the blockHash for multiple of our methods (for instance)
```
    /**
     * Returns block
     * @param {String} blockHash hex equivalent of a NEAR block hash
     */
    async routeEthGetBlockByHash([blockHash, returnTxObjects]) {
        try {
            assert(blockHash, 'Must pass in blockHash');
            blockHash = utils.hexToBase58(blockHash);

            ....
   
    }
```

Was there a decision made on this? When I try and change it back, it all starts breaking bc it isn't hex, I don't know why it was doing that before but such is the nature of developing in here, I'm finding

Reply:
so IIRC, the web3.js asserts that arguments are hex in some places. which is our method requires hex. you should check that tho

in general, I would _prefer_ if the b58 is used, to be consistent with other near tooling. but if it causes web3 integration issues, we should use hex. whichever it ends up being, we should be consistent if at all possible

Reply:
Yes, web3.js asserts hex in some places.

When possible, we use `b58`, but web3 often forces us to use hex. I'll have to go back through and note when and where that is. But if I remember, wherever there is hex, it is because web3.js asserted it to be. If web3 asserted a certain type, the arguments won't even make it to the Near Provider for us to convert back and forth.

Reply:
> on second thought, we probably want to retain the base58 blockhash for nearProtocol

I can't remember if web3 requires this to be hex, but I believe it does. I'll have to check.

Reply:
Flow:
1. User passes near web3 provider near hash in hex equivalent(`b58` to hex because web3 asserts for hex)
  - need to verify this is still the case
2. Near provider takes hex hash and converts it back to `b58` for near protocol to consume
3. Protocol passes results back in `b58`
4. Provider converts `b58` hash to hex to pass back to user
  - need verify web3 asserts hex type

Reply:
ok...in the context of this PR we have a `transactionObject` and a `receiptObject`. As if was before, blockHash was base58 and hex respectively. If we want them to be consistent, we have to go with Hex. Which I think `transactionObj` would be the ONLY outward facing base58 if we kept it that way. web3 is very stubborn and intertangled, huh lol.

Thanks for bearing with me. It's a bit tricky trying to get down all the formatting.

Reply:
if hex is good to go on all this, this is good for review.

Reply:
I see what you're saying now that we have not been consistent. I thought you meant we were returning for example `blockHash` as `b58` in one place and then `blockHash` as hex in another. You meant, either return all things in `b58` or in hex.

Reply:
yah. Just wondering for blockhash when base58 when hex. turns out...all hex all da time for near-web3-provider 😎